### ISSUE TITLE
pip uninstall afdko fails

### ISSUE THREAD

**Issue by miguelsousa** [adobe-type-tools/afdko#241]
Running `pip uninstall afdko` fails like this,
```python
Exception:
Traceback (most recent call last):
  File "/Users/msousa/fdk-test/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Users/msousa/fdk-test/lib/python2.7/site-packages/pip/commands/uninstall.py", line 76, in run
    requirement_set.uninstall(auto_confirm=options.yes)
  File "/Users/msousa/fdk-test/lib/python2.7/site-packages/pip/req/req_set.py", line 346, in uninstall
    req.uninstall(auto_confirm=auto_confirm)
  File "/Users/msousa/fdk-test/lib/python2.7/site-packages/pip/req/req_install.py", line 754, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/Users/msousa/fdk-test/lib/python2.7/site-packages/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/Users/msousa/fdk-test/lib/python2.7/site-packages/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 316, in move
    copy2(src, real_dst)
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 144, in copy2
    copyfile(src, dst)
  File "/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 96, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: '/Users/msousa/fdk-test/bin/buildcff2vf'
```

**Comment by cjchapman** [comment]
The `pip uninstall afdko` process seems to queue up both mixed case and lowercase versions of the same file names for removal, e.g.:
```
Uninstalling afdko-2.6.24:
...
  /Users/cchapman/Envs/fdk_install_test/bin/buildCFF2VF
...
  /Users/cchapman/Envs/fdk_install_test/bin/buildcff2vf
 ...
```
...even though only the mixed case version of the file name actually exists (i.e. `buildCFF2VF`).

It does that for all of these files:
```
buildCFF2VF
buildcff2vf
buildMasterOTFs
buildmasterotfs
checkOutlinesUFO
checkoutlinesufo
compareFamily
comparefamily
copyCFFCharstrings
copycffcharstrings
makeInstances
makeinstances
makeInstancesUFO
makeinstancesufo
stemHist
stemhist
```
…but it only looks for the mixed case version (and not the lowercase version) of:
```
mergeFonts
```

@anthrotype, any idea why this might be happening?

**Comment by frankrolf** [comment]
Is the case even important? Quick command-line tests don’t fail, for instance:
```
touch FILE.txt
rm file.txt
```
Maybe a file is removed twice and fails the 2nd time around?

**Comment by cjchapman** [comment]
@frankrolf, yes, that's exactly the problem, it's getting removed the first time, and then we get the failure that Miguel reported the second time.

**Comment by khaledhosny** [comment]
There are also case-sensitive file systems.

**Comment by frankrolf** [comment]
Oh yes, of course. `pip` also has to work on Windows, my apologizes.

**Comment by anthrotype** [comment]
windows is case insensitive too. Linux is case sensitive.
(I haven't looked into the issue with pip uninstall yet, sorry..)

**Comment by cjchapman** [comment]
The `pip show -f` command only shows the mixed case file names:
```
$ pip show -f afdko | grep -i "buildcff2vf"
  ../../../bin/buildCFF2VF
  afdko/Tools/SharedData/FDKScripts/buildCFF2VF.py
  afdko/Tools/SharedData/FDKScripts/buildCFF2VF.pyc
```

**Comment by miguelsousa** [comment]
@cjchapman do you know what actually gets installed? Is it camel case or all lowercase? Is that true for all platforms?

**Comment by cjchapman** [comment]
@miguelsousa, mixed case is what is actually installed on OS X. I haven't tried any other platforms.

**Comment by miguelsousa** [comment]
Then why is the process trying to uninstall the lowercased variants?

**Comment by cjchapman** [comment]
I don't know, that's what I'm trying to figure out.

**Comment by cjchapman** [comment]
This issue seems to be related: https://github.com/pypa/pip/issues/2834

I can confirm that when I step through the `pip` code in the debugger while uninstalling `afdko` that I hit this line in `ConfigParser.py`:
```python
                        optname = self.optionxform(optname.rstrip())
```
which brings us to this code, also in `ConfigParser.py`:
```python
    def optionxform(self, optionstr):
        return optionstr.lower()
```

**Comment by miguelsousa** [comment]
I was just cross referencing [your list](#issuecomment-358825839) with the contents of [setup.py](https://github.com/adobe-type-tools/afdko/blob/master/setup.py). You singled out `mergeFonts`. Does the same happen to `rotateFont`? If yes, then this issue seems to affect only [console scripts](https://github.com/adobe-type-tools/afdko/blob/master/setup.py#L160).

**Comment by cjchapman** [comment]
I missed it earlier.

For `mergeFonts` and `rotateFont` it only tries to remove the mixed case version.

**Comment by cjchapman** [comment]
I did the experiment of renaming all the target filenames in `entry_points['console_scripts']` in `setup.py` to lowercase and confirmed that I can cleanly `pip uninstall afdko` in that case (on my Mac, at least).

@miguelsousa, should I proceed with this workaround? Also what are your thoughts on renaming `mergeFonts` and `rotateFont` to be all lowercase for consistency?

**Comment by miguelsousa** [comment]
Yes, that seems fine to me.
As for renaming mergeFonts and rotateFont I'd leave them as is.

**Comment by anthrotype** [comment]
looks like it's a known issue of pip (the label says "awaiting PR")
https://github.com/pypa/pip/issues/3801

**Comment by anthrotype** [comment]
and this duplicate issue https://github.com/pypa/pip/issues/4771

**Comment by cjchapman** [comment]
@anthrotype, also https://github.com/pypa/pip/issues/2834, as I mentioned earlier

**Comment by cjchapman** [comment]
At least we know we're not alone. :-)

**Comment by miguelsousa** [comment]
Fixed by c79e754cd635ac5ea3aa247267828e0238b145ac
